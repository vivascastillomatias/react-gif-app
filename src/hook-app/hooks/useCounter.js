import { useState } from "react"

function useCounter( initialState = 10) {
    const [counter, setCounter] = useState(initialState)

    const increment = () => {
        setCounter(counter + 1)
    }

    const decrement = () => {
        setCounter(counter - 1)
    }

    const reset = () => {
        setCounter(initialState)
    }

    return {
        counter,
        increment,
        decrement,
        reset
    }
}

export default useCounter
