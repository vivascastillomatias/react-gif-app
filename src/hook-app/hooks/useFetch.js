import { useEffect, useRef, useState } from 'react'

function useFetch(url) {
    const [state, setState] = useState({ data: null, loading: true, error:null })

    const isMounted = useRef(true)

    useEffect(() => {
        return () => {
            isMounted.current = false;
        }
    }, [])

    useEffect(() => {
        setState({ data: null, loading: true, error:null })
        fetch(url)
            .then(resp => resp.json())
            .then(data => {
                    if (isMounted.current) {
                        setState({
                            data: data,
                            loading: false,
                            error:null
                            })
                    }else{
                        console.log('setstate no se llamo')
                    }
                });
    }, [url])

    return state;
}


export default useFetch
