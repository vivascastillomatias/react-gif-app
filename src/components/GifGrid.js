// import React, {useState, useEffect} from 'react'
import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
// import getGifs from '../helpers/getGifs';
import GifGridItem from './GifGridItem';

export default function GifGrid({category}) {

    // const [images, setImages] = useState([])

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn'>{category}</h3>
        { loading && <p className='animate__animated animate__flash'>loading...</p>}
        <div className='card-grid'>
            {
                images.map((img) => {
                    return <GifGridItem key={img.id} {...img}></GifGridItem>
                })
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
