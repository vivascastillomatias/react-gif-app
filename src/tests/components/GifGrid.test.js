import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {
    const category = 'Rap'
    
    test('Construcci칩n correcta del componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        }) //Simula el resultado que devolver치 el hook que ser치 almacenado en el componente probado

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan im치genes', () => {
        const gifs = [{
            id: 'id1',
            title: 'image1',
            url: 'urlImage1'
        },{ 
            id: 'id2',
            title: 'image2',
            url: 'urlImage2'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists() ).toBe(false)
        expect(wrapper.find('GifGridItem').length ).toBe(gifs.length)
    })
    
})
