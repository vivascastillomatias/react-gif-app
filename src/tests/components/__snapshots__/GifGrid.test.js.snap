// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Construcción correcta del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rap
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan imágenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rap
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="id1"
      key="id1"
      title="image1"
      url="urlImage1"
    />
    <GifGridItem
      id="id2"
      key="id2"
      title="image2"
      url="urlImage2"
    />
  </div>
</Fragment>
`;
